File: bubble.c
#include <stdio.h>

int main() {

    int n;
    scanf("%d", &n);

    int a[n];   // dynamic size array

    for(int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    // Bubble Sort with optimization
    for(int i = 0; i < n - 1; i++) {

        int swapped = 0;   // check if swap happened

        for(int j = 0; j < n - i - 1; j++) {
            if(a[j] > a[j + 1]) {
                int temp = a[j];
                a[j] = a[j + 1];
                a[j + 1] = temp;
                swapped = 1;
            }
        }//swapped

        // Print array after each pass
        for(int k = 0; k < n; k++) {
            printf("%d ", a[k]);
        }
        printf("\n");

        // If no swaps, array is already sorted
        if(swapped == 0) {
            break;
        }
    }

    return 0;
}

File: Bubble.java
import java.io.*;
import java.util.*;
import java.util.Scanner;
public class Bubble {

    public static void main(String[] args) {
        int n;
        Scanner s=new Scanner(System.in);
        n=s.nextInt();
        int a[]=new int[n];
        for(int i=0;i<n;i++) {
            a[i]=s.nextInt();
        }
        int temp;
        for(int i=0;i<n-1;i++) {
            for(int j=0;j<n-1-i;j++) {
                if(a[j]>a[j+1]) {
                    temp=a[j];
                    a[j]=a[j+1];
                    a[j+1]=temp;
                }
            }
            for(int k=0;k<n;k++) {
            System.out.print(a[k]+" ");
        }
        System.out.println();
        }
    }
}
